lib/src/main/java/org/altbeacon/beacon/startup/RegionBootstrap.java

```diff
@@ -34,24 +35,106 @@
 
     protected static final String TAG = "AppStarter";
     private BeaconManager beaconManager;
-    private BootstrapNotifier application;
+    private MonitorNotifier monitorNotifier;
+    private Context context;
     private List<Region> regions;
     private boolean disabled = false;
     private BeaconConsumer beaconConsumer;
     private boolean serviceConnected = false;
 
+    /**
+     * Constructor to bootstrap your Application on an entry/exit from a single region.
+     *
+     * @param context
+     * @param region
+     */
+    public RegionBootstrap(final Context context, final MonitorNotifier monitorNotifier, Region region) {
+        if (context.getApplicationContext() == null) {
+            throw new NullPointerException("Application Context should not be null");
+        }
+        beaconManager = BeaconManager.getInstanceForApplication(context.getApplicationContext());
+        this.context = context;
+        this.monitorNotifier = monitorNotifier;
+
+        this.monitorNotifier = new BootstrapNotifier() {
+            public Context getApplicationContext() {
+                return context.getApplicationContext();
+            }
+
+            public void didEnterRegion(Region region) {
+                monitorNotifier.didEnterRegion(region);
+            }
+
+            public void didExitRegion(Region region) {
+                monitorNotifier.didExitRegion(region);
+            }
+
+            public void didDetermineStateForRegion(int state, Region region) {
+                monitorNotifier.didDetermineStateForRegion(state, region);
+            }
+        };
+
+        regions = new ArrayList<Region>();
+        regions.add(region);
+        beaconConsumer = new InternalBeaconConsumer();
+        beaconManager.bind(beaconConsumer);
+        LogManager.d(TAG, "Waiting for BeaconService connection");
+    }
+
+    /**
+     * Constructor to bootstrap your Application on an entry/exit from multiple regions
+     *
+     * @param context
+     * @param regions
+     */
+    public RegionBootstrap(final Context context, final MonitorNotifier monitorNotifier, List<Region> regions) {
+        if (context.getApplicationContext() == null) {
```

Comment:
might be a good idea to just check for context == null, since that is the more likely case